### PR TITLE
feat(linter): add  jsx-a11y settings

### PR DESCRIPTION
**Pull request by msdlisper**
When we developed linter  for #1141 , we needed to configure some settings for `jsx-a11y`, which was not supported before, but I am trying to support it now.
like this:
```
fn config() -> serde_json::Value {
    serde_json::json!([2,{
        "ignoreNonDOM": true
    }])
}

fn settings() -> serde_json::Value {
    serde_json::json!({
        "jsx-a11y": {
            "components": {
                "Button": "button",
            }
        }
    })
}

let pass = vec![
    ("<Button />", Some(config()), Some(settings())),
];
```

### CODE REVIEW

**Comment by Boshen** [comment]
Thank you for the PR!

We now support reading from eslint config json! This means we can read values off 

https://www.npmjs.com/package/eslint-plugin-jsx-a11y#configurations

I think you can read these values into a struct (deserialized) and then add it to `LintContext`.

**Comment by msdlisper** [comment]
@Boshen I check out the new code just now,  let me read the settings from config json.

**Comment by msdlisper** [comment]
@Boshen Done :)

**Review by camc314 — APPROVED** [review]
Looks great to me! Thanks!

File: crates/oxc_linter/src/lib.rs [hunk 1]
```diff
@@ -33,10 +33,38 @@ pub use crate::{
...
+impl Default for LintSettings {
+    fn default() -> Self {
+        Self { jsx_a11y: JsxA11y { polymorphic_prop_name: None, components: HashMap::new() } }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct JsxA11y {
+    polymorphic_prop_name: Option<String>,
+    components: HashMap<String, String>,
```
Comment (Boshen) [inline crates/oxc_linter/src/lib.rs:50]: Let's use `use rustc_hash::FxHashMap` here.

**Review by Boshen — APPROVED** [review]
The hashmap can be done with a follow up PR.

**Pull request merged**